### 1. Introduction

<br>
<div style="font-size:15px;">
    <ul type="circle">
        <li>
            In this notebook we are going to deal with a supervised learning problem called regression.
        </li>
        <li>
            Regression is classical problem of Machine Learning task which considers the target variable to be continuous or real-valued.
        </li>
    </ul>
</div>


<div style="font-size:15px;">
    The collected data is from stock market and we are going to use linear regression on it.
</div>

<br>
<strong style="font-size:15px;">
    What is Linear Regression?
</strong>

### 2. Importing dataset

#### Looking for the dataset directory

In [1]:
from glob import glob
files = glob('*')

files

['data', 'Regression.ipynb']

<p style="font-size:15.5px;">
    The dataset is residing in <strong>data</strong> directory.
</p>

#### Getting file name

In [2]:
dataset = glob('data/*')

dataset

['data/stocks.csv']

#### Loading data into DataFrame using pandas.

In [3]:
import pandas as pd

stocks = pd.read_csv('data/stocks.csv')

#### Inspecting first 5 rows of dataset

In [4]:
stocks.head()

close                 date     high        id     low     open  ticker  \
0  1.13401  2018-12-05 04:24:00  1.13403  19703898  1.1338  1.13400  EURUSD   
1  1.13385  2018-12-05 04:25:00  1.13400  19703934  1.1338  1.13399  EURUSD   
2  1.13384  2018-12-05 04:26:00  1.13394  19703958  1.1337  1.13388  EURUSD   
3  1.13383  2018-12-05 04:27:00  1.13394  19703987  1.1337  1.13383  EURUSD   
4  1.13370  2018-12-05 04:28:00  1.13390  19704015  1.1336  1.13390  EURUSD   

   volume  
0     100  
1     118  
2      63  
3      51  
4      32

#### Number of rows and columns in our dataset

In [5]:
stocks.shape

(1440, 8)

<strong>Rows = 1440, Columns = 8</strong>

### 3. Exploratory Data Analysis (EDA)

<strong style="font-size:15.5px;">
    Inspecting the columns datatype and how many of them have null values.
</strong>


In [6]:
stocks.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1440 entries, 0 to 1439
Data columns (total 8 columns):
close     1440 non-null float64
date      1440 non-null object
high      1440 non-null float64
id        1440 non-null int64
low       1440 non-null float64
open      1440 non-null float64
ticker    1440 non-null object
volume    1440 non-null int64
dtypes: float64(4), int64(2), object(2)
memory usage: 90.1+ KB


#### Observations:

<br>
<div style="font-size:15px;">
    <ul type="circle">
        <li>
            None of the columns have null values, all the columns have entries to number of rows
        </li>
        <li>
            But there is a problem with the datatype of <strong>date</strong> column. It is of type <strong>object</strong> here but it should be of <strong>datetime</strong> type.
        </li>
    </ul>
</div>